khelo/lib/ui/flow/tournament/detail/tabs/tournament_detail_teams_tab.dart

```diff
@@ -5,47 +5,51 @@ import 'package:flutter_svg/svg.dart';
 import 'package:khelo/domain/extensions/context_extensions.dart';
 import 'package:khelo/domain/extensions/string_extensions.dart';
 import 'package:style/animations/on_tap_scale.dart';
-import 'package:style/button/bottom_sticky_overlay.dart';
-import 'package:style/button/primary_button.dart';
 import 'package:style/extensions/context_extensions.dart';
 import 'package:style/text/app_text_style.dart';
 
 import '../../../../../components/empty_screen.dart';
 import '../../../../../components/image_avatar.dart';
 import '../../../../../gen/assets.gen.dart';
 import '../../../../app_route.dart';
+import '../tournament_detail_view_model.dart';
 
 class TournamentDetailTeamsTab extends ConsumerWidget {
-  final List<TeamModel> teams;
   final Function(List<TeamModel>) onSelected;
 
   const TournamentDetailTeamsTab({
     super.key,
-    required this.teams,
     required this.onSelected,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    if (teams.isEmpty) {
-      return Stack(
-        children: [
-          EmptyScreen(
-            title: context.l10n.tournament_detail_teams_empty_title,
-            description: context.l10n.tournament_detail_teams_empty_description,
-            isShowButton: false,
-          ),
-          _stickyButton(context),
-        ],
+    final state = ref.watch(tournamentDetailStateProvider);
+    if (state.tournament!.teams.isEmpty) {
```

Comment:
why we're force unwrapping tournament? we haven't checked for non null before force unwrapping right?